pkg/metrics/wal/wal_test.go

```diff
@@ -452,14 +456,11 @@ func TestDBAllowOOOSamples(t *testing.T) {
 	}
 
 	require.NoError(t, app.Commit())
+
 	for _, metric := range payload {
-		for _, sa := range metric.samples {
-			// We want to set the timestamp to before. This should no longer trigger an out of order.
-			sa.ts = sa.ts - 10_000
-		}
-	}
-	for _, metric := range payload {
-		metric.Write(t, app)
```

Comment:
The Write helper was also trying to re-append exemplars, which now fails.

I've extracted this to a new helper method that includes the timestamp offset and does _not_ add exemplars.